utils/frame/try-runtime/cli/tests/create_snapshot.rs

```diff
@@ -0,0 +1,76 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#![cfg(unix)]
+#[cfg(feature = "try-runtime")]
+mod tests {
+	use assert_cmd::cargo::cargo_bin;
+	use regex::Regex;
+	use std::{
+		fs,
+		process::{self},
+		time::Duration,
+	};
+	use substrate_cli_test_utils as common;
+	use tokio::process::{Child, Command};
+
+	#[tokio::test]
+	async fn create_snapshot_works() {
+		// Build substrate so binaries used in the test use the latest code.
+		common::build_substrate(&["--features=try-runtime"]);
+
+		common::run_with_timeout(Duration::from_secs(60), async move {
+			fn create_snapshot(ws_url: &str) -> Child {
+				Command::new(cargo_bin("substrate"))
+					.stdout(process::Stdio::piped())
+					.stderr(process::Stdio::piped())
+					.args(&["try-runtime", "--runtime=existing"])
+					.args(&["create-snapshot", format!("--uri={}", ws_url).as_str()])
+					.kill_on_drop(true)
+					.spawn()
+					.unwrap()
+			}
+
+			// Start a node and wait for it to begin finalizing blocks
+			let mut node = common::KillChildOnDrop(common::start_node());
+			let ws_url = common::extract_info_from_output(node.stderr.take().unwrap()).0.ws_url;
+			common::wait_n_finalized_blocks(1, &ws_url).await;
+
+			// Try to create a snapshot.
+			let mut snapshot_creation = create_snapshot(&ws_url);
+			let re = Regex::new(r#".*writing snapshot of (\d+) bytes to .*"#).unwrap();
+			let matched =
+				common::wait_for_stream_pattern_match(snapshot_creation.stderr.take().unwrap(), re)
+					.await;
+
+			// Assert that the snapshot creation succeded.
+			assert!(matched.is_ok());
+
+			let snapshot = fs::read_dir(".")
+				.expect("Failed to read the current directory")
+				.filter_map(Result::ok)
+				.find(|entry| {
+					entry.path().is_file() &&
+						entry.path().extension().map(|ext| ext == "snap").unwrap_or(false)
```

Comment:
Can we make sure we clean up the file at the end of the test? 

I suggest specifying a name for it when you run `create-snapshot` so it can be easily identified for removal at the end.